File: BoP/lab5/lab_cpp/lab_5/lab_5/lab5.cpp
#include <iostream>
#include <vector>
#include <stdlib.h> 
#include <stdio.h>
#include <time.h> 
#include "materialPoint.h"
#include "Header.h"

using namespace std;

int main() {
	int k, n;
	float t;
	vector<vector<double>> coords;
	vector<double> distances;
	srand(time(NULL));
	cout << "Input k: "; cin >> k;
	cout << "Input n: "; cin >> n;
	cout << "Input t: "; cin >> t;
	
	vector<materialPointX> x;
	for (int i = 0; i < k; i++)
	{
		materialPointX* cl = new materialPointX(t);
		x.push_back(*cl);
		cout << "Coordinates: " << endl;  x[i].getCoordinates();
		for (int j = 0; j < i; j++)
		{
			double d = x[i].getDistance(&x[j]);
			vector<double> dc = {x[i].getX(), x[i].getY(), x[j].getX(), x[j].getY()};
			coords.push_back(dc);
			distances.push_back(d);
		}
	}

	vector<materialPointXY> xy;
	for (int i = 0; i < n; i++)
	{
		materialPointXY* cl = new materialPointXY(t);
		xy.push_back(*cl);
		cout << "Coordinates: " << endl;  xy[i].getCoordinates();
		for (int j = 0; j < i; j++)
		{
			double d = xy[i].getDistance(&xy[j]);
			vector<double> dc = { xy[i].getX(), xy[i].getY(), xy[j].getX(), xy[j].getY() };
			coords.push_back(dc);
			distances.push_back(d);
		}
	}

	for (int i = 0; i < k; i++)
	{
		for (int j = 0; j < n; j++)
		{
			double d = x[i].getDistance(&xy[j]);
			vector<double> dc = { x[i].getX(), x[i].getY(), xy[j].getX(), xy[j].getY() };
			coords.push_back(dc);
			distances.push_back(d);
		}
	}

	int ind = findMax(distances);
	cout << "Max distance: " << distances[ind] << endl;
	cout << "Between points with coordinates: (" << coords[ind][0] << ";" << coords[ind][1] << "), (" << coords[ind][2] << ";" << coords[ind][3] << ")" << endl;

}

File: BoP/lab5/lab_py/lab5.py
from material_point import MaterialPointX, MaterialPointXY

k = int(input("Input k: "))
n = int(input("Input n: "))
t = float(input("Input t: "))

coords = []
distances = []
xs = []
xys = []

for i in range(k):
    cl = MaterialPointX(t)
    xs.append(cl)
    print(cl)
    for j in range(i):
        dist = cl.get_distance(xs[j].get_x(), xs[j].get_y())
        coordinates = [cl.get_x(), cl.get_y(), xs[j].get_x(), xs[j].get_y()]
        coords.append(coordinates)
        distances.append(dist)

for i in range(n):
    cl = MaterialPointXY(t)
    xys.append(cl)
    print(cl)
    for j in range(i):
        dist = cl.get_distance(xys[j].get_x(), xys[j].get_y())
        coordinates = [cl.get_x(), cl.get_y(), xys[j].get_x(), xys[j].get_y()]
        coords.append(coordinates)
        distances.append(dist)

for i in range(k):
    for j in range(n):
        dist = xs[i].get_distance(xys[j].get_x(), xys[j].get_y())
        coordinates = [xs[i].get_x(), xs[i].get_y(), xys[j].get_x(), xys[j].get_y()]
        coords.append(coordinates)
        distances.append(dist)

ind = distances.index(max(distances))
print(f'Max distance: {distances[ind]}')
print(f'Between points with coordinates: ({coords[ind][0]}; {coords[ind][1]}), ({coords[ind][2]}; {coords[ind][3]})')
